purchase_allowed_product/models/product.py

```diff
@@ -19,18 +19,20 @@ def _search(
         count=False,
         access_rights_uid=None,
     ):
-        use_only_supplied_product = self.env.context.get("use_only_supplied_product")
-        if use_only_supplied_product:
+        if self.env.context.get("use_only_supplied_product"):
             restrict_supplier_id = self.env.context.get("restrict_supplier_id")
-            seller = self.env["res.partner"].browse(restrict_supplier_id)
-            seller = seller.commercial_partner_id
+            seller = (
+                self.env["res.partner"]
+                .browse(restrict_supplier_id)
+                .commercial_partner_id
+            )
             supplierinfos = self.env["product.supplierinfo"].search(
                 [("name", "=", seller.id)]
             )
             args += [
                 "|",
-                ("product_tmpl_id", "in", supplierinfos.mapped("product_tmpl_id").ids),
```

Comment:
This change is not correct. This will give `singleton error` in case of multiple supplierinfos found.

Reply:
Thank you for review. I agree with you that the code `supplierinfos.product_tmpl_id.ids` would be incorrect in v12 but since v13, multi-relational field access is supported and works like a mapped https://www.odoo.com/documentation/13.0/reference/orm.html#map. This change should not cause an error even supplierinfos is a recordset.

Reply:
I'm afraid not:

![Selección_005](https://user-images.githubusercontent.com/7165771/99415994-aa5aa600-28f8-11eb-993c-fb05270f07c3.png)

It seems their initial intention was to allow it and then go back again to previous behavior, but they don't update documentation.

@Feyensv can you please confirm and rectify documentation if so?

Reply:
Since 13, mapped is implicit, but only for relational fields, the ORM will raise when accessing another type of field on multiple records at once:
```python
partners = users.partner_id
user_names = users.mapped('name') # necessary because len(users) > 1
# implicit mapped also works for long relational walkthrough:
currencies = users.partner_id.company_id.currency_id
currency_names = users.partner_id.company_id.currency_id.mapped('name')
currency_rates = users.partner_id.company_id.currency_id.rate_ids
```

Reply:
Thanks for the clarification, Victor. I think the documentation can be clarified for avoiding this misunderstanding, as it's a bit tricky and for me was not clear, or is it only me?

Reply:
It is documented, but maybe not highlighted enough in the big ORM page ^^.
![image](https://user-images.githubusercontent.com/19364378/99432316-46da7380-290c-11eb-9ea0-238f9d2d796c.png)

Reply:
Yeah, I mean something more explicit. What about?

```
Since v13, you can chain relational fields (many2one, one2many, many2many) without the use mapped when having multi recordsets, but not for other field types.

records.partner_id  # == records.mapped('partner_id')
records.partner_id.bank_ids  # == records.mapped('partner_id.bank_ids')
records.partner_id.mapped('name')  # name field is not relational, so it needs mapped
```

Reply:
@Feyensv do you see correct my proposal for changing the documentation? Should I put it in a PR?